server/channels/app/ldap.go

```diff
@@ -60,6 +61,19 @@ func (a *App) TestLdapConnection(rctx request.CTX, settings *model.LdapSettings)
 		"ent.ldap.disabled.app_error", nil, "", http.StatusNotImplemented)
 }
 
+func (a *App) TestLdapFilters(rctx request.CTX, settings *model.LdapSettings) ([]model.LdapFilterTestResult, *model.AppError) {
+	license := a.Srv().License()
+	ldapI := a.LdapDiagnostic()
+
+	// NOTE: normally we would test (*a.Config().LdapSettings.Enable || *a.Config().LdapSettings.EnableSync),
+	// but we want to allow sysadmins to test the connection without enabling and saving the config first.
+	if ldapI != nil && license != nil && *license.Features.LDAP {
+		return ldapI.RunTestFilters(rctx, settings)
+	}
+
+	return nil, model.NewAppError("TestLdapConnection", "ent.ldap.disabled.app_error", nil, "", http.StatusNotImplemented)
```

Comment:
Any reason to keep the happy path scoped and defaulting to an error?

Reply:
Not really, I was just following the existing pattern.  I'll look at when I do the endpoint combining now.